beacon_node/beacon_chain/src/kzg_utils.rs

```diff
@@ -19,6 +34,50 @@ pub fn validate_blob<E: EthSpec>(
     kzg.verify_blob_kzg_proof(&kzg_blob, kzg_commitment, kzg_proof)
 }
 
+/// Validate a batch of `DataColumnSidecar`.
+pub fn validate_data_columns<'a, E: EthSpec, I>(
+    kzg: &Kzg,
+    data_column_iter: I,
+) -> Result<(), KzgError>
+where
+    I: Iterator<Item = &'a Arc<DataColumnSidecar<E>>> + Clone,
+{
+    let cells = data_column_iter
+        .clone()
+        .flat_map(|data_column| data_column.column.iter().map(ssz_cell_to_crypto_cell::<E>))
+        .collect::<Result<Vec<_>, KzgError>>()?;
+
+    let proofs = data_column_iter
+        .clone()
+        .flat_map(|data_column| {
+            data_column
+                .kzg_proofs
+                .iter()
+                .map(|&proof| Bytes48::from(proof))
+        })
+        .collect::<Vec<_>>();
+
+    let column_indices = data_column_iter
+        .clone()
+        .flat_map(|data_column| {
+            let col_index = data_column.index;
+            (0..data_column.column.len()).map(move |_| col_index)
```

Comment:
```suggestion
            data_column.column.iter().map(move |_| col_index)
```

Reply:
Fixed in https://github.com/sigp/lighthouse/pull/6212/commits/9b69f1db20e36a8cad27f0bee8583cb84305f82e